### PR TITLE
fix(DB/SAI/Creature): Kyle's Gone Missing! Quest credit

### PR DESCRIPTION
**Pull request by febaut**
## Changes Proposed:
-  Adding a new SMART SCRIPT line to Kyle the Frenzied (Entry: 23616) that handles better the Kyle's Gone Missing! Quest credit
-  Removing the current quest credit trigger from Kyle the Friendly's action list. (entryorguid = 2362200 ) (This one only completes quest for closest player)

## Issues Addressed:
 --- 
 
- Closes https://github.com/chromiecraft/chromiecraft/issues/2727
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10023

## Tests Performed:

-  Applied the DB changes and did the How to Test steps again.

## How to Test the Changes:

find kyle

1. .q a 11129
2. .additem 33009
3. .go c 24762
4. place meat
5. get some distance (10 yards was the max limit according to the issue)

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Shineslippers** [comment]
Hi, there @febaut . When you're adding `#Issues Addressed:` please chekout line `Original report:` in AC report body if it's exist and add to Addressed it. In this case there is `Original report: chromiecraft/chromiecraft#2727`

File: data/sql/updates/pending_db_world/rev_1641437831789184769.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1641437831789184769');
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=2362200 AND `source_type`=9 AND `id`=2 AND `link`=0;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=23616 AND `source_type`=0 AND `id`=3 AND `link`=0;
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1641437831789184769.sql:4]: what is this deleting? id 3 line does not exsist in our db in the smart scripts

File: data/sql/updates/pending_db_world/rev_1641437831789184769.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1641437831789184769');
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`=2362200 AND `source_type`=9 AND `id`=2 AND `link`=0;
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1641437831789184769.sql:3]: we have 5 ids for this entry (0 to 5), removing just id 2 will break the id chain. please adjust

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1641437831789184769.sql (hunk 1 above)
Comment (febaut) [reply data/sql/updates/pending_db_world/rev_1641437831789184769.sql:4]: Yeah is the requested Delete before Insert.

File: data/sql/updates/pending_db_world/rev_1641437831789184769.sql (hunk 2 above)
Comment (febaut) [reply data/sql/updates/pending_db_world/rev_1641437831789184769.sql:3]: Oh yeah, didn't think about it. Will fix it. Thank u.

File: data/sql/updates/pending_db_world/rev_1641437831789184769.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+UPDATE `acore_world`.`smart_scripts` SET `id`='2' WHERE  `entryorguid`=2362200 AND `source_type`=9 AND `id`=3 AND `link`=0;
+UPDATE `acore_world`.`smart_scripts` SET `id`='3' WHERE  `entryorguid`=2362200 AND `source_type`=9 AND `id`=4 AND `link`=0;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1641437831789184769.sql:5]: Remove acore_world and don't wrap integers in quotes. Thanks :)

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by Nyeriah** [comment]
Hey, sorry for taking long to reply to this. I wanted to check this carefully and I didn't have the time to look into it until now.

I have just one question about this approach. If the problem was that the action list used the nearby player as target, why are we not fixing the target so it targets the invoker instead of adding the credit on creature spawn?

**Comment by febaut** [comment]
@Nyeriah Hey, sorry. Didn't see your comment. 

Checking the Script logic it shows that Kyle the frenzied stores the target `Kyle the Frenzied - On Spellhit 'Lunch for Kyle' - Store Targetlist` an that was not used before. Also  Kyle the Friendly is summoned by a player's action, that's why action invoker as target won't work.

So, Kyle the frenzied stores the player that places the meal, and then when he spawns the friendly one, the player gets the credit.

**Comment by Lynethris** [comment]
Tested and work.

**Comment by acidmanifesto** [comment]
Is this PR abandon or will it be corrected?

**Comment by febaut** [comment]
Done

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**